applications/DEMApplication/python_scripts/DEM_material_test_script.py

```diff
@@ -336,35 +368,8 @@ def BtsSkinDetermination(self):
         self.Procedures.KratosPrintInfo("Finished computing the skin of the BTS specimen..." + "\n")
 
     def PrepareDataForGraph(self):
-
-        prepare_check = [0,0,0,0]
-        self.total_check = 0
-
-        for smp in self.rigid_face_model_part.SubModelParts:
-            if smp[IDENTIFIER] == "TOP":
-                self.top_mesh_nodes = smp.Nodes
-                prepare_check[0] = 1
-            if smp[IDENTIFIER] == "BOT":
-                self.bot_mesh_nodes = smp.Nodes
-                prepare_check[1] = 1
-
-        for smp in self.spheres_model_part.SubModelParts:
-            if smp[IDENTIFIER] == "TOP":
-                self.top_mesh_nodes = smp.Nodes
-                prepare_check[2] = -1
-
-            if smp[IDENTIFIER] == "BOT":
-                self.bot_mesh_nodes = smp.Nodes
-                prepare_check[3] = -1
-
-        for it in range(len(prepare_check)):
-
-            self.total_check += prepare_check[it]
-
-        if math.fabs(self.total_check) != 2:
-
-            self.Procedures.KratosPrintWarning(" ERROR in the definition of TOP BOT groups. Both groups are required to be defined, they have to be either on FEM groups or in DEM groups")
-
+        self.CheckCorrectNameOfPlates()
```

Comment:
Is this done at the begining? We don't want to wait many time steps to make sure the data is correctly set...